### PR TITLE
Media Indicator and Link position bugs

### LINKED ISSUE #685: Media Indicator overlap bug
**Describe the bug**
On viewports above 400px, any time the text grid item is taller than the image one, the media indicator drops off the bottom of the image

**Expected behaviour**
The media indictaor stays perfectly ontop of the image on >400px, then entirely below it.

**Screenshots**
image

**Possible fix:**
Add a wrapper around the contents of ImageGridItem [here](https://github.com/bbc/psammead/blob/latest/packages/components/psammead-story-promo/src/index.jsx#L249-L253) with "position: relative". This means that the media indicator isn't positioned to the bottom of the grid item, but the new div, which's size is dictated by the children (the image)

**Additional context**

### LINKED ISSUE #2071: Story promo overlays the page
**Describe the bug**
On mobile devices such as iPhoneX, the viewport is overlayed by a link.
This behaviour is caused by the `::before` tag that set overlay over the single story promo, however that seems to leak out over the entire page.

**To Reproduce**

1. Go to https://www.stage.bbc.co.uk/yoruba on a mobile device
2. Click anywhere on the page without scrolling
3. You will be taken to an article (the article is the last article shown on the page)

**Expected behavior**
Click on a story, you will be taken to the story.

**Screenshots**
image
image
image

**Smartphone:**

- Device: iPhoneX, iPhone5S, Samsung Galaxy S5
- Browser: Firefox, Chrome

**Additional context**
Checked with some other devices, like OnePlus6 and Pixel2, and the page works as expected.

Bug spotted on iPhoneX (real device). This was also checked in Chrome dev tools simulator with iPhone5S and Samsung Galaxy S5 settings. 
Checked with Pixel2 settings in Chrome dev tools as well.

### PR DESCRIPTION
**Pull request by dr3**
Resolves https://github.com/bbc/simorgh/issues/2071
Resolves #685

**Overall change:** Updates story promo to resolve 2 positioning bugs

**Code changes:**

- Adds some position: relative were needed

Testing #2071
Add a blank area to the story to ensure the link doesnt spill out. To do this I added a [padding component](https://github.com/bbc/psammead/blob/009e311cfa2884c649a43b412e9661e5a908f0ce/packages/components/psammead-story-promo/src/index.stories.jsx#L50-L54) and inserted this into the story [like this](https://github.com/bbc/psammead/blob/009e311cfa2884c649a43b412e9661e5a908f0ce/packages/components/psammead-story-promo/src/index.stories.jsx#L71) in browserstack, if I tap the blue area, it doesnt click the link, unlike before.

Testing #685
Edit the headline text in the dev tools to be reeeeeeally long, the media indicator shouldn't drop with it.

General regression of top story and media indicator behaviour needed :) 

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-story-promo/src/index.jsx [hunk 1]
```diff
@@ -109,6 +107,10 @@ const ImageGridItem = styled.div`
   }
 `;
 
+const ImageContentsWrap = styled.div`
```
Comment (DenisHdz) [inline packages/components/psammead-story-promo/src/index.jsx]: I'm being a bit picky here but, can this be either `ImageContentsWrapper` or `ImageContent`?

Comment (dr3) [reply packages/components/psammead-story-promo/src/index.jsx]: done :)

**Review by DenisHdz — APPROVED** [review]
LGTM 👍🏻

**Review by ghost — APPROVED** [review]

**Review by j-pendlebury — APPROVED** [review]

**Comment by staylos92** [comment]
LGTM 👍

**Pull request merged**